mergify_engine/rules/__init__.py

```diff
@@ -278,29 +389,41 @@ async def create(
         self = cls(rules)
 
         for rule in self.rules:
-            for condition in rule.conditions:
-                for attrib in self.TEAM_ATTRIBUTES:
-                    # mypy thinks can take only Callable and not Coroutine
+            for attrib in self.TEAM_ATTRIBUTES:
+                for condition in rule.conditions.walk():
                     condition.partial_filter.value_expanders[
                         attrib
                     ] = functools.partial(  # type: ignore[assignment]
                         live_resolvers.teams, ctxt
                     )
 
-            await rule.conditions.eval(ctxt.pull_request)
-
-            for condition in rule.conditions:
-                if hide_rule and condition.evaluation_error:
+            await rule.conditions(ctxt.pull_request)
+
+            # NOTE(sileht):
+            # In the summary, we display rules in four groups:
+            # * rules where all attributes match -> matching_rules
+            # * rules where only BASE_ATTRIBUTES match (filter out rule written for other branches) -> matching_rules
+            # * rules that won't work due to a configuration issue detected at runtime (eg team doesn't exists) -> faulty_rules
+            # * rules where only BASE_ATTRIBUTES don't match (mainly to hide rule written for other branches) -> ignored_rules
+            categorized = False
+
+            # FIXME(sileht): We may need to revisit this boolean
```

Comment:
can you write why?

Reply:
I removed it, I fixed the issue in another PR